server/whitelist/whitelist.go

```diff
@@ -0,0 +1,43 @@
+package whitelist
+
+import (
+	"strings"
+	"sync"
+
+	"github.com/rancher/rancher/pkg/settings"
+)
+
+var (
+	Proxy = ProxyList{
+		RWMutex:        sync.RWMutex{},
+		whitelistProxy: map[string]bool{},
+	}
+)
+
+type ProxyList struct {
+	sync.RWMutex
+	whitelistProxy map[string]bool
+}
+
+func (p *ProxyList) Get() []string {
+	p.RLock()
+	p.RUnlock()
```

Comment:
Hi, are you missing here the defer? Because unlock directly after the lock statement doesn't make much sense. For the performance it should be better to not use the defer here, instead just write the unlock at the end of the func.

Reply:
sorry I missed it. WIll add it back, thanks!